### PR TITLE
initial implementation for static outputs in cloud

### PR DESCRIPTION
**Pull request by mslynch**
Important note – for cloud, appId now refers to the content id instead of the actual application id, and takes the format `lucid:content:<id>` due to domain modeling differences between Cloud and Connect.

### CODE REVIEW

File: R/ide.R [hunk 1]
```diff
@@ -114,7 +114,7 @@ showRstudioSourceMarkers <- function(basePath, lint) {
 
 # https://github.com/rstudio/rstudio/blob/ee56d49b0fca5f3d7c3f5214a4010355d1bb0212/src/gwt/src/org/rstudio/studio/client/rsconnect/ui/RSConnectDeploy.java#L699
 
-getAppById <- function(id, account, server, hostUrl) {
+getAppById <- function(id, account, server, hostUrl, contentId = NULL) {
```
Comment (mslynch) [inline R/ide.R:117]: The IDE will need to pass content id here.

File: R/deployApp.R [hunk 2]
```diff
@@ -152,6 +152,7 @@ deployApp <- function(appDir = getwd(),
                       appName = NULL,
                       appTitle = NULL,
                       appId = NULL,
+                      contentId = NULL,
```
Comment (omar-rs) [inline R/deployApp.R]: `contentId` needs to be added to the docs above (otherwise the build will fail, I think)

Comment (omar-rs) [reply R/deployApp.R]: is `contentId` always `NULL` in the case of `deployApp` (only relevant for static outputs)?  If so, we don't need to add it as a new arg for this function.

**Comment by mbaynton** [comment]
I'm noticing that the IDE regards the application ID as a string. I'm wondering if it might be better / clearer / require touching fewer projects if we just persisted static content in the rsconnect data files as `lucid:content:[id]` and used that inside the lucid client's `getApplication` method to look up our content entity instead.

**Comment by mbaynton** [comment]
More specifically, I'm interested in playing with an arrangement where the `appId` field in the `.dcf` file persisting where an output has been published to gets set to `lucid:content:[id]`

**Comment by mslynch** [comment]
> I'm noticing that the IDE regards the application ID as a string. I'm wondering if it might be better / clearer / require touching fewer projects if we just persisted static content in the rsconnect data files as `lucid:content:[id]` and used that inside the lucid client's `getApplication` method to look up our content entity instead.

I had been thinking that approach but I recall having some reason not to do it... I can't remember why but it seems like a better idea after writing this PR. I'll give it a try.

**Comment by mslynch** [comment]
> I had been thinking that approach but I recall having some reason not to do it... I can't remember why but it seems like a better idea after writing this PR. I'll give it a try.

Update: Implemented this. I think it looks a lot better.

File: R/deployApp.R (hunk 2 above)
Comment (mslynch) [reply R/deployApp.R]: took this out.

**Review by omar-rs — APPROVED** [review]

File: R/deployments.R [hunk 3]
```diff
@@ -92,7 +92,7 @@ saveDeployment <- function(recordDir,
     account = target$account,
     server = target$server,
     hostUrl = hostUrl,
-    appId = application$id,
+    appId = paste("lucid", "content", application$content_id, sep = ":"),
```
Comment (mbaynton) [inline R/deployments.R]: This looks like it occurs regardless of server type. I think we shouldn't put "lucid" in there when referring to an entity on a connect server.

**Review by mbaynton — COMMENTED** [review]
I tested publishing to a connect server using this and got some failures, we'll need to test that case more.

```
── Preparing for deployment ────────────────────────────────────────────────────
✔ Deploying "rmarkdown-site-2" to "mike.baynton@colorado.rstudio.com"
ℹ Creating application on server...
✔ Created application with id 
ℹ Bundling 64 files: 'about.html', 'index.html', 'site_libs/bootstrap-3.3.5/css/bootstrap-theme.css', 'site_libs/bootstrap-3.3.5/css/bootstrap-theme.css.map', 'site_libs/bootstrap-3.3.5/css/bootstrap-theme.min.css', 'site_libs/bootstrap-3.3.5/css/bootstrap.css', 'site_libs/bootstrap-3.3.5/css/bootstrap.css.map', 'site_libs/bootstrap-3.3.5/css/bootstrap.min.css', 'site_libs/bootstrap-3.3.5/css/cerulean.min.css', 'site_libs/bootstrap-3.3.5/css/cosmo.min.css', 'site_libs/bootstrap-3.3.5/css/darkly.min.css', 'site_libs/bootstrap-3.3.5/css/flatly.min.css', 'site_libs/bootstrap-3.3.5/css/fonts/Lato.ttf', 'site_libs/bootstrap-3.3.5/css/fonts/LatoBold.ttf', 'site_libs/bootstrap-3.3.5/css/fonts/LatoItalic.ttf', 'site_libs/bootstrap-3.3.5/css/fonts/NewsCycle.ttf', 'site_libs/bootstrap-3.3.5/css/fonts/NewsCycleBold.ttf', 'site_libs/bootstrap-3.3.5/css/fonts/OpenSans.ttf', …, 'site_libs/navigation-1.1/sourceembed.js', and 'site_libs/navigation-1.1/tabsets.js'
✔ Created 1,285,784b bundle
ℹ Uploading bundle...
Error in `POST()`:
! <https://colorado.posit.co/rsc/__api__> failed with HTTP status 404
404 page not found
```

File: R/deployments.R (hunk 3 above)
Comment (mslynch) [reply R/deployments.R]: Pushed a commit to fix this.

**Comment by mslynch** [comment]
Closing this PR in favor of https://github.com/rstudio/rsconnect/pull/825 (the main repo has secrets we need for all the tests to pass).

**Pull request closed without merging**